BluetoothLEExplorer/BluetoothLEExplorer/Models/ObservableBluetoothLEDevice.cs

```diff
@@ -722,12 +728,13 @@ private async void BluetoothLEDevice_NameChanged(BluetoothLEDevice sender, objec
 
         private async void BluetoothLEDevice_GattServicesChanged(BluetoothLEDevice sender, object args)
         {
-            await Windows.ApplicationModel.Core.CoreApplication.MainView.CoreWindow.Dispatcher.RunTaskAsync(async () =>
+            await Windows.ApplicationModel.Core.CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(
+                Windows.UI.Core.CoreDispatcherPriority.Normal,
+                () =>
             {
-                // If we received a service change.  If we are in the middle of discovery services,
-                // we can ignore the service change.
-                await GetAllPrimaryServices(IsPaired ? BluetoothCacheMode.Cached : BluetoothCacheMode.Uncached);
+                GattServicesChangedInstances += 1;

```

Comment:
>GattServicesChangedInstances += 1; [](http://example.com/codeflow?start=16&length=34)

Since we set this on every event, does that mean that this counter will always show at least `1`, even for a device that hasn't had a "real" service change?

Reply:
It seems to happen only when first connecting to a perepherial. When we hit refresh it actually just reads 0 (until a service change comes along).

Reply:
Yeah, the API internally keeps track of the services so it's only changed when there is a change. So you will likely see 1 during the initial population and more everytmne there is a change detected.

Reply:
Still a little unfortunate that this will show 1 by default, since it may trick a user into hitting refresh an extra time to get "into sync". But not the end of the world.